### DISCUSSION TITLE
Surprising parameters when saving an PNG image

### DISCUSSION THREAD

**Discussion by pstein** [tannerhelland/PhotoDemon#527] [General]
I loaded a PNG image and performed some edit actions.
Then I clicked on menu "Save" or "Save As" to save the edited image.

Much to my surprise a save dialog pops up with  some (for me) unknown storing parameters:

before/after
1:1/fit
compression level
compression optimization
embed background color

Very strange. I have never seen these parameters for PNG in other image editors.

What do the mean in detail?
I know Compression level only form JPG files. 
PNG files should be always compressed by one and only method.
I am confused.

Assume there are such parameters. Does PD recognize them at loading time and use 
them (=the same) automatically at save time?

**Comment by VladimirMarkelov** [comment]
PNG uses various compression algorithms that are called levels: https://cimage.se/doc/png-compression-settings

**Comment by pstein** [comment]
Ok.
And what are the differences between 

before/after

1:1/fit

**Comment by VladimirMarkelov** [comment]
Even for lossless compression you can change(see "advanced settings in the preview), e.g, the number of colors(decrease to 256) or make PNG grayscale. It may make PNG look very different comparing to the original. So, 

before = original image, after = image that you get after saving with parameters defined by you.

1;1/fit = what to show in the preview window: show the image with real size(you have to drag the image to see certain part of the image)/fit the whole image to the preview window(sometimes it is too small to notice the difference)
